### DISCUSSION TITLE
How to enable Pulsar non-persistent topic?

### DISCUSSION THREAD

**Discussion by xiaobingxia-at** [milvus-io/milvus#37279] [Q&A and General discussion] [answered]
Hello, does any one know how to enable the Pulsar Non-Persistent topic?
Because somehow we don't want the data to be landed on disk without encryption, we decided to try out the Pulsar Non-Persistent topic which is completely in memory.
But I didn't find any document about how to enable that using milvus operator yaml. Does anyone have any insight?

Thanks!

**Comment by yhmo** [comment]
Milvus officially provide these configurations for pulsar: https://milvus.io/docs/configure_pulsar.md
I don't think there is a configuration for non-persistent topic. Maybe need a workaround.

Wait @LoveEachDay to comment.

**Comment by LoveEachDay** [comment, marked as answer]
@xiaobingxia-at Milvus does not support producing or consuming messages from pulsar non-persistent topics.

**Comment by xiaobingxia-at** [comment]
Thank you! @LoveEachDay 
The main motivation is because we don't want data landed on disk without encryption.
I assume the "Bring Your Own Key" feature could be a solution for this. Is "Bring Your Own Key" going to be released at Milvus 2.5?

**Reply by xiaofan-luan** [reply]
unfortunately BYOK will only be part of our enterprise edition so the answer is no.

and also by using non persistent topic you may lose data when pulsar crash.

I think we can can support user to implement their own encrption key reader like pulsar https://pulsar.apache.org/docs/4.0.x/security-encryption/.

any one want to work on this feature?

**Reply by xiaofan-luan** [reply]
I think we can first support encrption all the data as a whole
